### PR TITLE
[ENH] Process text improve

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Add `new_column_names` argument to add a new column or columns, if a Series/dataframe is returned.
- Add `merge_frame` argument to merge dataframe back to original dataframe, if a dataframe is returned.
-

**This PR improves #694 .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - flake8 checking
    - running the test suite
    - docs build

Once done, please check off the check-box above.

If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [ ] Check to ensure that test coverage covers the lines of code that you have added.
    - [ ] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: tests/functions/test_process_text.py [hunk 1]
```diff
@@ -4,46 +4,174 @@
 from pandas.testing import assert_frame_equal
 
 
-def test_str_split():
-    "Test wrapper for Pandas ``.str.split()`` method."
-
-    df = pd.DataFrame(
+@pytest.fixture
+def test_df():
```
Comment (ericmjl) [inline tests/functions/test_process_text.py]: Minor nitpick, which I think might affect pytest usage: fixtures commonly don't have a `test_` prefix. Could we call this fixture `process_text_df`, and rename them accordingly below please?

**Review by ericmjl — APPROVED** [review]
Only one change request here, @samukweku. Other than that, pre-approving!

File: tests/functions/test_process_text.py (hunk 1 above)
Comment (samukweku) [reply tests/functions/test_process_text.py]: Thanks @ericmjl i will work on the suggestions

**Comment by ericmjl** [comment]
h/t @samukweku!

**Pull request merged**